pkg/checker/checker.go

```diff
@@ -114,7 +114,7 @@ func check(definition *api.Definition) func() error {
 	return func() error {
 		resp, err := doStatusRequest(definition, true)
 		if err != nil {
-			return err
+			return fmt.Errorf("%s health check endpoint %s is unreachable", definition.Name, definition.HealthCheck.URL)
```

Comment:
But now we do not know why it failed - it can be invalid endpoint, or timeout error, or connectivity problem etc.

Reply:
Yes actually that was the intention, hide all the error variations and provide a generic one, so API consumers don't need to be familiar with go http errors. Perhaps provide a better error message?